econplayground/main/views.py

```diff
@@ -221,12 +223,23 @@ class GraphEmbedPublicMinimalView(
     template_name = 'main/graph_embed_public_minimal.html'
 
 
-class GraphDeleteView(UserPassesTestMixin, DeleteView):
+class GraphDeleteView(LoginRequiredMixin, CohortInstructorMixin, DeleteView):
     model = Graph
-    success_url = '/'
 
-    def test_func(self):
-        return user_is_instructor(self.request.user)
+    def get_context_data(self, **kwargs):
+        ctx = super(GraphDeleteView, self).get_context_data(**kwargs)
+        ctx.update({
+            'cohort': self.cohort,
+        })
+        return ctx
+
+    def get_success_url(self):
+        messages.add_message(
+            self.request, messages.SUCCESS,
+            '<strong>{}</strong> has been deleted.'.format(self.object.name),
```

Comment:
The Graph object has a `title` attribute - not a `name`. Running the graph delete process with this change causes an exception - we should add a test for this process.

Reply:
I will add the failing test for Graph deletion here.